### ISSUE TITLE
Question about using robot state information during RDT2-FM post-training

### ISSUE THREAD

**Issue by JinmingM** [thu-ml/RDT2#10]
Hi, thanks for your great work!

I have a question regarding the post-training process in RDT2-FM.
It seems that the robot’s proprioceptive information (state) is not used during pre-training — is that correct?
https://github.com/thu-ml/RDT2/blob/main/rdt/dataset.py#L132

If so, would it be possible to include the robot’s state information during the post-training stage?
Also, should the state vector strictly have a 20-dimensional or can it be customized?
https://github.com/thu-ml/RDT2/blob/main/configs/rdt/post_train.yaml#L11

Looking forward to your reply!

Thanks again for the excellent work.

**Comment by JinmingM** [comment]
Thanks for your reply! I have one more question:

It seems that the file **`umi_normalizer_wo_downsample_indentity_rot.pt`** includes the robot’s proprioceptive information (state).
Could you please explain **how this normalizer file was generated**?
Is there any **corresponding code in the repository** that processes the dataset to compute and save this normalizer?

For example, the file contains structures like:

```
LinearNormalizer(
  (params_dict): ParameterDict(
      (action): Object of type: ParameterDict
      (robot0_eef_pos): Object of type: ParameterDict
      (robot0_eef_rot_axis_angle): Object of type: ParameterDict
      (robot0_gripper_width): Object of type: ParameterDict
      (robot1_eef_pos): Object of type: ParameterDict
      (robot1_eef_rot_axis_angle): Object of type: ParameterDict
      (robot1_gripper_width): Object of type: ParameterDict
      (robot0_eef_pos_wrt1): Object of type: ParameterDict
      (robot0_eef_rot_axis_angle_wrt1): Object of type: ParameterDict
      (robot1_eef_pos_wrt0): Object of type: ParameterDict
      (robot1_eef_rot_axis_angle_wrt0): Object of type: ParameterDict
  )
)
```

I’m curious about how these normalization parameters (mean, std, etc.) were computed and saved.

Thanks again for your time and clarification!

**Comment by JinmingM** [comment]
Thanks for your reply!

**Comment by Li-Yun-star** [comment]
The link provided in https://ml.cs.tsinghua.edu.cn/~lingxuan/rdt2/umi_normalizer_wo_downsample_indentity_rot.pt
 is no longer available. Could you please provide an updated link, or share the file via Baidu Disk or Google Drive?

**Comment by MichaelWu99-lab** [comment]
> Hi,
> 
> I’m using the UMI setup to collect a dual-arm folding-clothes dataset (~1000 episodes). This dataset works well with Diffusion Policy, so I believe the data quality is good. However, when I use RDT2-FM for post-training, the learned policy cannot successfully complete the task.
> 
> Could you please help me confirm whether my post-training procedure is correct?
> 
> 1. I converted the UMI zarr dataset into WebDataset format with image resolution (384, 384),  and generated normalization using UMI’s Python scripts.
>    My state and action spaces are defined as:
>    
>    * **state**: (pos + rot6d representation + gripper) × 2 arms, obs horizon = 1
>    * **action**: (pos + rot6d representation + gripper) × 2 arms, action chunk = 24, relative representation
> 2. My post-train script is as follows (abbreviated for clarity):
> 
> ENV_TYPE="post_train"
> WANDB_PROJECT="rdt2-action-expert-UMI20"
> VISION_LANGUAGE_MODEL_NAME_OR_PATH="/robot/code/RDT2/rdt2_models/RDT2-VQ/"
> ACTION_MODEL_NAME_OR_PATH="/robot/code/RDT2/rdt2_models/RDT2-FM/"
> 
> PYTHONPATH="/robot/code/RDT2" accelerate launch rdt/main.py \
>     --deepspeed="scripts/zero1.json" \
>     --config_path="./configs/rdt/${ENV_TYPE}.yaml" \
>     --pretrained_vision_language_model_name_or_path=$VISION_LANGUAGE_MODEL_NAME_OR_PATH \
>     --pretrained_model_name_or_path=$ACTION_MODEL_NAME_OR_PATH \
>     --output_dir=$OUTPUT_DIR \
>     --webdataset_config=$WDS_CONFIG_FILE \
>     --train_batch_size=12 \
>     --sample_batch_size=2 \
>     --max_train_steps=100000 \
>     --learning_rate=1e-4 \
>     --mixed_precision="bf16" \
>     --image_aug \
>     --report_to=wandb
> Could you please check if this setup aligns with the intended post-train procedure for RDT2-FM? Is there any additional step needed to adapt the dual-arm dataset or state/action dimensions?
> 
> Thank you very much!

Hello, may I ask if your problem has been resolved? I have encountered the same confusion as you. Maybe we can have a conversation about it.
